ChildProject/annotations.py

```diff
@@ -0,0 +1,287 @@
+        'MA4': 'MAL',
+        'MA5': 'MAL',
+        'MAE': 'ELE',
+        'MC1': 'OCH',
+        'MC2': 'OCH',
+        'MC3': 'OCH',
+        'MC4': 'OCH',
+        'MC5': 'OCH',
+        'MI1': 'OCH',
+        'MOT*': 'FEM',
+        'OC0': 'OCH',
+        'UC1': 'OCH',
+        'UC2': 'OCH',
+        'UC3': 'OCH',
+        'UC4': 'OCH',
+        'UC5': 'OCH',
+        'UC6': 'OCH'
+    }
+
+
+    def __init__(self, project):
+        self.project = project
+        self.annotations = None
+        self.errors = []
+
+        if not isinstance(project, ChildProject):
+            raise ValueError('project should derive from ChildProject')
+
+        project.read()
+
+        index_path = os.path.join(self.project.path, 'annotations/annotations.csv')
+        if not os.path.exists(index_path):
+            open(index_path, 'w+').write(','.join([c.name for c in self.INDEX_COLUMNS]))
+
+        errors, warnings = self.read()
+
+    def read(self):
+        table = IndexTable('input', path = os.path.join(self.project.path, 'annotations/annotations.csv'), columns = self.INDEX_COLUMNS)
+        self.annotations = table.read()
+        errors, warnings = table.validate()
+        return errors, warnings
+
+    def validate(self):
+        errors, warnings = [], []
+
+        for annotation in self.annotations.to_dict(orient = 'records'):
+            segments = IndexTable(
+                'segments',
+                path = os.path.join(self.project.path, 'annotations', annotation['annotation_filename']),
+                columns = self.SEGMENTS_COLUMNS
+            )
+
+            segments.read()
+            res = segments.validate()
+            errors += res[0]
+            warnings += res[1]
+
+        return errors, warnings
+        
+
+    def load_textgrid(self, filename):
+        path = os.path.join(self.project.path, 'raw_annotations', filename)
+        textgrid = pympi.Praat.TextGrid(path)
+
+        segments = []
+        for tier in textgrid.tiers:
+            for interval in tier.intervals:
+                tier_name = tier.name.strip()
+
+                if tier_name == 'Autre':
+                    continue
+
+                if interval[2] == "":
+                    continue
+
+                segment = {
+                    'segment_onset': "{:.3f}".format(interval[0]),
+                    'segment_offset': "{:.3f}".format(interval[1]),
+                    'speaker_id': tier_name,
+                    'ling_type': interval[2] if interval[2] else "",
+                    'speaker_type': self.SPEAKER_ID_TO_TYPE[tier_name] if tier_name in self.SPEAKER_ID_TO_TYPE else 'NA',
+                    'vcm_type': 'NA',
+                    'lex_type': 'NA',
+                    'mwu_type': 'NA',
+                    'addresseee': 'NA',
+                    'transcription': 'NA'
+                }
+
+                segments.append(segment)
+
+        return pd.DataFrame(segments)
+
+    def load_eaf(self, filename):
```

Comment:
It passes the tests, but can you see any issue from looking at the code ?
@alecristia

Reply:
'segment_onset': "{:.3f}".format(interval[0]) <-- note that in theory VTC should return 2 digits (sometimes it does not, but I don't know why, since frames are 10 ms long)

Also, this is a good place to fix entries by humans that go beyond the period they needed to code -- so lines 137-8 would fix onset/offsets that go beyond the coding period.

Otherwise, it looks good!

Reply:
Thanks!